reindexer/reindex_worker/src/main/scala/uk/ac/wellcome/platform/reindex/reindex_worker/services/RecordReader.scala

```diff
@@ -35,7 +35,11 @@ class RecordReader @Inject()(
           totalSegments = reindexJob.totalSegments
         )
 
-      recordsToReindex: List[HybridRecord] = results.map(extractRecord)
+      recordsToReindex = reindexJob.maxRecordsPerSegment match {
+        case None => results.map(extractRecord)
+        case Some(maxRecordsPerSegment) =>
+          results.take(maxRecordsPerSegment).map(extractRecord)
```

Comment:
What’s the reason for this change? I think it’ll cause the reindexer to start dropping records, because anything beyond `maxRecordsPerSegment` will be lost.

Is this useful for something in the reporting pipeline?